great_expectations/validator/metric_configuration.py

```diff
@@ -33,32 +38,34 @@ def __str__(self):
         return self.__repr__()
 
     @property
-    def metric_name(self):
+    def metric_name(self) -> str:
         return self._metric_name
 
     @property
-    def metric_domain_kwargs(self):
+    def metric_domain_kwargs(self) -> IDDict:
         return self._metric_domain_kwargs
 
     @property
-    def metric_value_kwargs(self):
+    def metric_value_kwargs(self) -> IDDict:
         return self._metric_value_kwargs
 
     @property
-    def metric_domain_kwargs_id(self):
+    def metric_domain_kwargs_id(self) -> str:
         return self.metric_domain_kwargs.to_id()
 
     @property
-    def metric_value_kwargs_id(self):
+    def metric_value_kwargs_id(self) -> str:
         return self.metric_value_kwargs.to_id()
 
     @property
-    def metric_dependencies(self):
-        return self._metric_dependencies
+    def metric_dependencies(self) -> Dict[str, MetricConfiguration]:
+        return self._metric_dependencies or {}
 
     @metric_dependencies.setter
-    def metric_dependencies(self, metric_dependencies) -> None:
-        self._metric_dependencies = metric_dependencies
+    def metric_dependencies(
+        self, value: Optional[Dict[str, MetricConfiguration]]
+    ) -> None:
+        self._metric_dependencies = value
```

Comment:
This point is totally orthogonal to the main point of this PR but I thought I'd bring it up since I've noticed this pattern a lot in our codebase.

The pattern is where we use a `@property` with a setter but we don't actually protect or calculate anything.
I think it just adds noise to the code.

Consider the 2 equivalent class definitions for the class `Square` which takes a `length` and `color`.
Both behave exactly the same but one is almost `20` lines while the other is only `4`.

```python
class Square:
    def __init__(self, length: int, color: str):
        self._length = length
        self._color = color

    @property
    def length(self) -> int:
        return self._length

    @length.setter
    def length(self, length: int):
        self._length = length

    @property
    def color(self) -> str:
        return self.color

    @color.setter
    def color(self, color: str):
        self._color = color

# the class below is functionally equivalent 

class Square:
    def __init__(self, length: int, color: str):
        self.length = length
        self.color = color
```

To be clear, this isn't an anti-property take. They are [very useful and powerful](https://pabloariasal.github.io/2018/11/25/python-descriptors/), but I think we are overusing in many situations.

Below are examples where IMO they do make sense.
1. We want to prevent modifying an attribute (`length`)
2. We want to apply some logic during getting or settings
  a. Like computing a value like `area` from length.
  b. Applying validation during setting `color`.

```python
class Square:
    def __init__(self, length: int, color: str):
        self._length = length
        self._color = color

    @property
    def length(self) -> int:
        return self._length

    @property
    def area(self) -> int:
        return self._length * self._length

    @property
    def color(self) -> str:
        return self.color

    @color.setter
    def color(self, color: str):
        if color not in ["red", "blue", "yellow"]:
            raise ValueError
        self._color = color
```

Reply:
@Kilo59 By the way, I have removed the `setter` from `MetricConfiguration.metric_dependencies()` property (per @donaldheppner 's observation).  Thanks.